src/bot/conversation.py

```diff
@@ -5,79 +5,81 @@
+def initializeConversation(message, bot, newConversation):
+    global user_known_words, chat_history
+    user_id = get_or_create_user_id_in_DB(str(message.from_user.id), message.from_user.username, message.from_user.first_name, message.from_user.last_name)
+    all_words = get_all_words(user_id)
+    user_known_words = []
+    for i in range(len(all_words)):
+        user_known_words.append(str(all_words[i][1]))
+    user_known_words = str(user_known_words) #might be misunderstandable the fact that in random terms i call user_known_words the list of german words + translations while here only german words
+    print(f"\nDictionary from user {message.from_user.first_name} and dictionary is: containing {len(all_words)} words\n")
+    _manageConversation(message, bot, newConversation, [])
 
 
-def callOpenAI(message, bot, newConversation):
+def _manageConversation(message, bot, newConversation, chat_history):
     """
-    Function to start a conversation with the user and get responses from OpenAI
+    Function to start or continue a conversation with the user and get responses from OpenAI
     
     args:
     message: the message object from the user
     bot: the bot object to send the message
     newConversation: a boolean indicating if this is a new conversation or not
+    chat_history: the conversation history array
     
     returns:
-    userConversation: the conversation history
+    chat_history: the conversation history
     """
     try:
         if newConversation:
-            userConversation = []
-            userConversation.append({"role": "system", "content": "You are a bot that helps students to learn German. You need to have simple conversations, with short sentences, using only present tense. You will mainly use terms from the dictionary in the TermsList file, as these are the words the student knows. \nHere is the list of the terms: " + german_words})
+            chat_history = []
+            chat_history.append({"role": "system", "content": f"You are a bot that helps students to learn German. You need to have simple conversations, with short sentences, using only present tense. You will mainly use terms from the dictionary in the TermsList file, as these are the words the student knows. \nHere is the list of the terms: {user_known_words}"})
             assistantMessage = bot.reply_to(
                 message, f"Hallo, ich kann dir helfen zu Deutsch zu sprechen! 🇩🇪" + '\n' + "Remember you can end the conversation anytime by typig `end`"
             )
-            userConversation.append({"role": "assistant", "content": assistantMessage.text})
+            chat_history.append({"role": "assistant", "content": assistantMessage.text})
             bot.register_next_step_handler(
-                message, lambda msg: llmresponse(msg, client, bot)
+                message, lambda msg: _get_llm_response(msg, chat_history, client, bot)
```

Comment:
It looks like you don't need to put this into the if-else statement. This bot `register_next_step_handler` gets called the same way here and a few lines down below in the else statement. You can clean the code by moving it down and delete the duplicate.